src/java.security.jgss/share/classes/sun/security/jgss/wrapper/GSSNameElement.java

```diff
@@ -292,9 +292,17 @@ public boolean isAnonymousName() {
     }
 
     public void dispose() {
-        if (pName != 0) {
-            cStub.releaseName(pName);
+        if (pName != 0 && cleanable != null) {
+            cleanable.clean();
             pName = 0;
         }
     }
+
+    private static Runnable disposerFor(GSSLibStub stub, long pName) {
+        return () -> {
+            if (stub != null && pName != 0) {
+                stub.releaseName(pName);
+            }
+        };
+    }
```

Comment:
There seem to be a potential optimization here, since you don't even need to create a Cleanable in the case where stub is null or pName is 0? Same for earlier duplication of the same code.